airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py

```diff
@@ -60,12 +52,12 @@ def _prune_futures(self, futures: List[Future[Any]]) -> None:
 
         for index in reversed(range(len(futures))):
             future = futures[index]
-            optional_exception = future.exception()
-            if optional_exception:
-                exception = RuntimeError(f"Failed reading with error: {optional_exception}")
-                self._stop_and_raise_exception(exception)
 
             if future.done():
+                optional_exception = future.exception()
```

Comment:
only check if the future is already done

Reply:
Worth adding a comment to explain why. Based on the [doc](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.exception), `If the call hasn’t yet completed then this method will wait up to timeout seconds.`